### PR TITLE
WebDeviceInputSystem: Added workaround for MacOS Chromium based Browsers

### PR DESCRIPTION
**Pull request by PolygonalSun**
There was an issue found with Chromium based browsers on MacOS where if the cursor left the browser window after clicking on the canvas with a mouse button other than left click, pointer capture will be lost and the canvas will still think that there's a button down.  Because this is a Chromium issue, the only thing that we can do is to create a workaround until it's fixed.  This workaround will catch the `lostpointercapture` event generated and release the button, if necessary.

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/14210/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14210/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14210/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/14210/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/14210/merge
https://gui.babylonjs.com/?snapshot=refs/pull/14210/merge
https://nme.babylonjs.com/?snapshot=refs/pull/14210/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/14210/merge#BCU1XR#0

File: packages/dev/core/src/DeviceInput/webDeviceInputSystem.ts [hunk 1]
```diff
@@ -765,11 +775,19 @@ export class WebDeviceInputSystem implements IDeviceInputSystem {
             }
         };
 
+        // Workaround for MacOS Chromium Browsers for lost pointer capture bug
+        this._pointerMacOSChromeOutEvent = (evt) => {
+            if (this._usingMacOS && !this._usingSafari && evt.buttons > 1) {
```
Comment (RaananW) [inline packages/dev/core/src/DeviceInput/webDeviceInputSystem.ts:780]: firefox is also available for mac. would be better to filter if it is mac and chromium (and not mac and not safari). Will also be safer for other browsers.

**Review by RaananW — COMMENTED** [review]
Not the biggest fan of such a custom solution integrated in the constructor TBH. 

Maybe move the function registration to a polyfill/browser dependent file, outside of ths web device input system context?
You could also register the event in case the environment is mac + chromium, instead of registering it on all browsers. This way the if in the function itself will be much simpler as well.

**Comment by PolygonalSun** [comment]
> is this issue tracked by chromium?

Yeah, it looks like it's been on their radar for a while but hasn't gotten any traction:
https://bugs.chromium.org/p/chromium/issues/detail?id=1131348

File: packages/dev/core/src/DeviceInput/webDeviceInputSystem.ts (hunk 1 above)
Comment (PolygonalSun) [reply packages/dev/core/src/DeviceInput/webDeviceInputSystem.ts]: Okay, I modified it to check to be more precise

**Comment by bjsplat** [comment]
Visualization tests for webgl2 have failed. If some tests failed because the snapshots do not match, the report can be found at

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14210/merge/testResults/webgl2/index.html

If tests were successful afterwards, this report might not be available anymore.

**Comment by bjsplat** [comment]
Visualization tests for webgl1 have failed. If some tests failed because the snapshots do not match, the report can be found at

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14210/merge/testResults/webgl1/index.html

If tests were successful afterwards, this report might not be available anymore.

**Comment by PolygonalSun** [comment]
> Not the biggest fan of such a custom solution integrated in the constructor TBH.
> 
> Maybe move the function registration to a polyfill/browser dependent file, outside of ths web device input system context? You could also register the event in case the environment is mac + chromium, instead of registering it on all browsers. This way the if in the function itself will be much simpler as well.

I can do that, lemme make that change now

**Comment by bjsplat** [comment]
Visualization tests for webgl1 have failed. If some tests failed because the snapshots do not match, the report can be found at

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14210/merge/testResults/webgl1/index.html

If tests were successful afterwards, this report might not be available anymore.

**Comment by bjsplat** [comment]
Visualization tests for webgl2 have failed. If some tests failed because the snapshots do not match, the report can be found at

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14210/merge/testResults/webgl2/index.html

If tests were successful afterwards, this report might not be available anymore.

**Review by sebavan — APPROVED** [review]

**Review by RaananW — APPROVED** [review]

**Pull request merged**